universal-application-tool-0.0.1/app/services/applicant/ApplicantServiceImpl.java

```diff
@@ -189,10 +189,14 @@ private void stageUpdates(
       throws UnsupportedScalarTypeException, PathNotInBlockException {
     ImmutableSet.Builder<Path> questionPaths = ImmutableSet.builder();
     for (Update update : updates) {
+      // Check existence of the non-array path, which is how scalars are indexed.
       ScalarType type =
           blockDefinition
-              .getScalarType(update.path())
-              .orElseThrow(() -> new PathNotInBlockException(blockDefinition, update.path()));
+              .getScalarType(update.path().withoutArrayReference())
```

Comment:
(blocking): can this be removed if checkbox question path ends with "[]"?

Reply:
I pushed this into `BlockDefinition` - the path from the form will end with `[index]`, not `[]`, so we still need custom logic to handle this case whether or not the scalar path itself has `[]`